### PR TITLE
Try removing replace directives from go.mod.

### PR DESCRIPTION
**Pull request by cjwagner**
I'm not sure if this will work, but it would be really nice to get rid of the `replace` directives in our go.mod file so that we can `go get` and `go install` packages from this repo.

/cc @listx

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -35,6 +21,10 @@ require (
 	github.com/bazelbuild/buildtools v0.0.0-20200922170545-10384511ce98
 	github.com/blang/semver/v4 v4.0.0
 	github.com/bwmarrin/snowflake v0.0.0
+	// Upstream is unmaintained. This fork introduces two important changes:
```
Comment (listx) [inline go.mod:24]: Can you check that running `make update-go-deps` does not erase these comments?

File: go.mod [hunk 2]
```diff
@@ -35,6 +21,10 @@ require (
...
+	// * We log an error if writing a cache key fails (e.G. because disk is full)
```
Comment (airbornepony) [inline go.mod:25]: Nit: e.g.

File: go.mod (hunk 1 above)
Comment (cjwagner) [reply go.mod:24]: Just tried locally and it didn't change any files. :+1:

**Comment by listx** [comment]
Looks like just a handful of unit tests fail. In your next update, apart from fixing the broken tests, could you also link to https://github.com/golang/go/issues/44840#issuecomment-1651863470 in your commit message description?

**Comment by cjwagner** [comment]
I just removed the failing unit test cases for the `golint` plugin. I can't find a changelog to prove it, but it seems like the linter behavior changed slightly.

**Comment by cjwagner** [comment]
Looks like the update I made to the AKS sdk is affected by the breaking changes. We'll need to follow [the migration guide here](https://github.com/Azure/azure-sdk-for-go/blob/main/documentation/MIGRATION_GUIDE.md) to remove that replace directive.
@droslean I'm trying to remove the replace directive added here: https://github.com/kubernetes/test-infra/commit/9be1ab79c2696e56b604a3304d5e7be911f018dd#diff-33ef32bf6c23acb95f5902d7097b7a1d5128ca061167ec0716715b0b9eeaa5f6R8
Do you have the bandwidth to help with updating kubetest to use the new version of this SDK? I'll try to take a look later to see how involved it is.

**Comment by droslean** [comment]
> Looks like the update I made to the AKS sdk is affected by the breaking changes. We'll need to follow [the migration guide here](https://github.com/Azure/azure-sdk-for-go/blob/main/documentation/MIGRATION_GUIDE.md) to remove that replace directive. @droslean I'm trying to remove the replace directive added here: [9be1ab7#diff-33ef32bf6c23acb95f5902d7097b7a1d5128ca061167ec0716715b0b9eeaa5f6R8](https://github.com/kubernetes/test-infra/commit/9be1ab79c2696e56b604a3304d5e7be911f018dd#diff-33ef32bf6c23acb95f5902d7097b7a1d5128ca061167ec0716715b0b9eeaa5f6R8) Do you have the bandwidth to help with updating kubetest to use the new version of this SDK? I'll try to take a look later to see how involved it is.

@cjwagner Sure. I'll try to update kubetest.

**Comment by cjwagner** [comment]
Thanks @droslean! 
I've pulled out the problematic AKS stuff into a separate commit so that we can get most of this merged.

File: go.mod (hunk 2 above)
Comment (cjwagner) [reply go.mod:27]: Oops forgot to fix this. I've addressed it in the next PR.

**Pull request merged**